### PR TITLE
feat(manuals): update manuals

### LINKED ISSUE #39: GitHub page and example projects
**What is the issue about**:
GitHub pages should be set up along with two example projects:
- hello world
- todo app

**What happened**:
Nothing happened, I just figure that a more out-reaching stance this repo shall take.

**What you expected to happen**:
More open source usage, more tractions, etc.

**Anything else we need to know**:

- [x] skeleton #40 
- [ ] add status badge and add *fork me at github button* 
- [ ] choose a better theme
- [x] create `CNAME` and apply domain

### LINKED ISSUE #154: framework overview in readme is not succinct enough
/kind feature

See [Rocket's](https://rocket.rs/overview/) for a better example. Ideally there should be a one pager that explains the core usage.

### PR DESCRIPTION
**Pull request by kdada**
**What this PR does / why we need it**:

This PR add Chinese manuals and update docs.

This PR does not finish all manuals. These works will be done in the future PRs.

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #154 
Fixes #39 

Reference to #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: manuals/zh-hans/guides/client.md [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	Title   string `json:"title"`
+	Content string `json:"content"`
+}
+```
+
+生成的客户端是版本化的，版本在 nirvana.yaml 中定义。API 依赖的结构体都会被提取出来并生成到 `types.go` 文件中，方便客户端使用。
+
+## 使用客户端
+
+客户端的使用非常简单，只需要创建客户端，然后调用相应的 API 函数即可：
```
Comment (ddysher) [inline manuals/zh-hans/guides/client.md]: Please give an end to end guide on how to use client, i.e. a runnable example, and give ur best practice here (should we vendor the client? or copy/past? or use a separate repository, etc).

File: manuals/zh-hans/guides/project.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
+# 项目结构和初始化
+
+## 创建项目
+Nirvana 创建项目非常简单，不过在创建项目之前，首先使用下载安装 Nirvana 的命令行和其他依赖工具：
```
Comment (ddysher) [inline manuals/zh-hans/guides/project.md]: s/首先使用下载安装/首先需要下载安装

File: manuals/zh-hans/plugins/logger.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Logger 插件
```
Comment (ddysher) [inline manuals/zh-hans/plugins/logger.md]: 在 `快速入门` 下添加一个`常用插件`，方便快速了解并用上？

File: manuals/zh-hans/guides/api.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+### Nirvana Definition
+在 Nirvana 中，所有的 API 都是通过 Descriptor 和 Definition 进行描述的。首先看一个 `List Messages` 的 API 定义：
+```go
+// 在使用 nirvana init 创建的标准项目结构中，这个文件位于 pkg/api/v1/descriptors/message.go
+
+func init() {
+	register([]def.Descriptor{{
+        // Path 定义了 API 路径
+		Path:        "/messages",
```
Comment (zjj2wry) [inline manuals/zh-hans/guides/api.md]: 这里没有对齐，mkdown 渲染的时候也是没有对齐的，是 gofmt 导致吗

File: manuals/zh-hans/guides/api.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+    // 这个字符串去掉空格后会作为生成客户端时的函数名，因此请确保这个字符串是有意义的。
+	Summary:     "List Messages",
+    // 详细描述这个 API 的用途。
+	Description: "Query a specified number of messages and returns an array",
+    // 业务函数
+	Function:    message.ListMessages,
+    // 对应业务函数的参数信息。用于告知 Nirvana 从请求的那一部分取得数据，然后传递给业务函数。
+	Parameters: []def.Parameter{
+		{
+            // 参数来源
```
Comment (zjj2wry) [inline manuals/zh-hans/guides/api.md]: ditto

File: manuals/zh-hans/guides/api.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+	messages := make([]Message, count)
+	for i := 0; i < count; i++ {
+		messages[i].ID = i
+		messages[i].Title = fmt.Sprintf("Example %d", i)
+		messages[i].Content = fmt.Sprintf("Content of example %d", i)
+	}
+	return messages, nil
+}
+```
+可以看到，业务函数不关心参数的来源和类型转换，而返回值也不用关心如何写如到请求的响应里。
```
Comment (zjj2wry) [inline manuals/zh-hans/guides/api.md]: 如何--> 然后？

File: manuals/zh-hans/guides/api.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+
+## 给项目添加一个 API
+接下来我们给项目增加一个 API，用于获取一条消息：
+```go
+func init() {
+	register([]def.Descriptor{{
+		Path:        "/messages",
+		Definitions: []def.Definition{listMessages},
+	}, {
+        // 获取一条消息的 Descriptor。
```
Comment (zjj2wry) [inline manuals/zh-hans/guides/api.md]: 对齐

File: manuals/zh-hans/guides/api.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+	return &Message{
+		ID:      id,
+		Title:   "This is an example",
+		Content: "Example content",
+	}, nil
+}
+```
+
+添加 API 之后，编译运行。然后访问 `http://localhost:8080/api/v1/messages/100`，即可获得结果（默认情况下都是 json 类型）：
+```
```
Comment (zjj2wry) [inline manuals/zh-hans/guides/api.md]: json

File: manuals/zh-hans/guides/client.md [hunk 9]
```diff
@@ -0,0 +1,158 @@
...
+func MustNewClient(cfg *rest.Config) *Client {
+	client, err := NewClient(cfg)
+	if err != nil {
+		panic(err)
+	}
+	return client
+}
+
+// GetMessage return a message by id.
+func (c *Client) GetMessage(ctx context.Context, message int) (message1 *Message, err error) {
```
Comment (zjj2wry) [inline manuals/zh-hans/guides/client.md]: message1 --> message

File: manuals/zh-hans/guides/client.md [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+## 使用客户端
+
+客户端的使用非常简单，只需要创建客户端，然后调用相应的 API 函数即可：
+```go
+func main() {
+	cli := client.MustNewClient(&rest.Config{
+		Scheme: "http",
+		Host:   "localhost:8080",
+	})
+	msgs, err := cli.V1().ListMessages(context.Background(), 10)
```
Comment (zjj2wry) [inline manuals/zh-hans/guides/client.md]: can not set API group NOW?

File: manuals/zh-hans/guides/project.md [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+# 项目负责人
+contacts:
+- name: nobody
+  email: nobody@nobody.io
+  description: Maintain this project
+# 项目 API 版本信息，用于区分不同版本的 API
+# 用于文档和客户端生成
+versions:
+  # 版本名称
+- name: v1
```
Comment (zjj2wry) [inline manuals/zh-hans/guides/project.md]: 一个版本一个项目？

File: manuals/zh-hans/plugins/profiling.md [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+
+前缀路径 `/debug/pprof` 可以通过 Path Configurer 修改。
+
+插件 Configurer：
+- Disable() nirvana.Configurer
+  - 关闭插件
+- Path(path string) nirvana.Configurer
+  - 设置路径前缀，默认值为 `/debug/pprof`
+- Contention(enable bool) nirvana.Configurer
+  - 启用阻塞事件采样
```
Comment (zjj2wry) [inline manuals/zh-hans/plugins/profiling.md]: 这是什么，加个链接？

File: manuals/zh-hans/guides/project.md [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+  # 版本描述
+  description: The v1 version is the first version of this project
+  # 版本规则
+  rules:
```
[the remark this answers is not in the document]
Comment (kdada) [reply manuals/zh-hans/guides/project.md]: Yes. `nirvana.yaml` is only for API docs and client generation. It won't impact server running.

File: manuals/zh-hans/guides/client.md (hunk 10 above)
Comment (kdada) [reply manuals/zh-hans/guides/client.md]: For a micro service, it should only contain a minimal group of APIs. API groups are aggregated by API gateway.

File: manuals/zh-hans/guides/project.md (hunk 11 above)
Comment (kdada) [reply manuals/zh-hans/guides/project.md]: One project can contains may versions of APIs. May be `/api/v1`, `/api/v2`. `versions` 
 describes all versions of this project.

File: manuals/zh-hans/plugins/profiling.md (hunk 12 above)
Comment (kdada) [reply manuals/zh-hans/plugins/profiling.md]: I removed this configurer from docs. It also will be removed from `profiling` plugin. It's an old concept before golang 1.10.

File: manuals/zh-hans/guides/client.md (hunk 9 above)
Comment (kdada) [reply manuals/zh-hans/guides/client.md]: No. Conflicted.

File: manuals/zh-hans/guides/api.md (hunk 8 above)
Comment (kdada) [reply manuals/zh-hans/guides/api.md]: ?

**Pull request merged**